client/state/wp-data/calypso-registry-plugin.js

```diff
@@ -0,0 +1,92 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { mapValues, without } from 'lodash';
+
+const calypsoRegistryPlugin = calypsoStore => registry => {
+	const namespaces = {};
```

Comment:
The fact that the plugin needs to maintain **its own** registry strikes me as a bit cumbersome. The desired state of things would be that there is one registry that allows to register all types of stores that provide a required interface (i.e., expose `selectors` and `actions`), no matter what the implementation is.

The current `@wordpress/data` registry seems to favor only one particular implementation of a store and doesn't accept any other one.

Using `@wordpress/data` public API to select a store now has an unexpected result:
```js
@import { select } from '@wordpress/data';
select( 'calypso' ).colorSchemePreference();
```
It will crash because the `calypso` store is not there: it's in another, private registry.

Reply:
I don't recommend using the global `select, dispatch, subscribe` exported functions. These are mainly for the WP context and even there we're thinking about "deprecating" them (won't be easy) in favor of always relying on the `registry`object (`registry.select/dispatch/subcribe`) and the HoCs.

Reply:
Still, I believe that a single registry should be able to rule them all. Is there any obstacle that would prevent that? I noticed that the P2 integration needs to patch an existing store's resolver (i.e., something that's not public API) -- that would be harder with a registry that doesn't know about store's internals.

Reply:
Yes, I think a single registry is capable of handing everything. Right now, we use the "default registry" in WordPress packages (and Gutenberg) which may drive Calypso to use the same default registry for now, but I'm hoping we can move away from the default registry in the future and each consumer would have to decide what to include in its registry.

```js
import { createRegistry } from '@wordpress/data';

import coreStoreConfig from '@wordpress/core-data';
import editorStoreConfig from '@wordpress/editor';
import calypsoStoreConfig from '../calypso';

export createRegistry( { 
  'core': coreStoreConfig,
  'core/editor': editorStoreConfig,
  'calypso': calypsoStoreConfig,
} ); 
```